src/main/java/fruitshop/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,22 @@
+package fruitshop.service.impl;
+
+import fruitshop.model.Storage;
+import fruitshop.service.ReportGenerator;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    public static final String HEADER = "fruit,quantity";
+    public static final String COMA = ",";
+    private final Storage storage = new Storage();
+
+    @Override
+    public String getReport() {
+        Map<String, Integer> fruitStorage = storage.getStorage();
+        StringBuilder report = new StringBuilder(HEADER).append(System.lineSeparator());
+        for (Map.Entry<String, Integer> fruit : fruitStorage.entrySet()) {
+            report.append(fruit.getKey()).append(COMA).append(fruit.getValue())
+                    .append(System.lineSeparator());
+        }
+        return report.toString();
```

Comment:
The `getReport` method is directly accessing the `storage` which makes it tightly coupled with the `Storage` class. Consider using a service or repository interface to interact with the storage, which will make it easier to change the underlying storage implementation in the future.